open-hackathon-server/src/hackathon/user/user_manager.py

```diff
@@ -326,15 +297,15 @@ def __oauth_login(self, provider, context):
 
         user = self.__get_existing_user(openid, provider)
         if user is not None:
-            self.db.update_object(user,
-                                  provider=provider,
-                                  name=context.get("name", user.name),
-                                  nickname=context.get("nickname", user.nickname),
-                                  access_token=context.get("access_token", user.access_token),
-                                  avatar_url=context.get("avatar_url", user.avatar_url),
-                                  last_login_time=self.util.get_now(),
-                                  login_times=user.login_times + 1,
-                                  online=1)
+            user.update(
+                provider=provider,
+                name=context.get("name", user.name),
+                nickname=context.get("nickname", user.nickname),
+                access_token=context.get("access_token", user.access_token),
+                avatar_url=context.get("avatar_url", user.avatar_url),
```

Comment:
Be careful of `avatar_url`, once complained by someone. `avatar_url` set by user will be override in login. We may want to differentiate those two.